vertx-context/src/main/java/io/smallrye/common/vertx/VertxContext.java

```diff
@@ -0,0 +1,88 @@
+     *
+     * @param context the context, must not be {@code null}
+     * @return the given context if it's a duplicated context, or a new duplicated context using the given one as
+     *         <em>root</em>.
+     */
+    public static Context getOrCreateDuplicatedContext(Context context) {
+        if (isDuplicatedContext(context)) { // Also checks that the context is not null
+            return context;
+        } else {
+            return ((ContextInternal) context).duplicate();
+        }
+    }
+
+    /**
+     * Gets or creates a duplicated context.
+     *
+     * If this method is called from a non-Vert.x thread (so, there is no current context), it creates a new (event loop)
+     * context and a duplicated context (and returns the duplicated context).
+     *
+     * If this method is called from a Vert.x thread (so, there is a current context), and if the current context is
+     * already a duplicated context, it returns the current duplicated context.
+     *
+     * It this method is called from a Vert.x thread (so, there is a current context), and if the current context is not
+     * a duplicated context, it creates a new duplicated context, using the current context as <em>root</em> context.
+     *
+     * @param vertx the Vert.x instance to use to create the context if needed. Must not be {@code null}
+     * @return the current context if it's a duplicated context, or a new duplicated context.
+     * @see #getOrCreateDuplicatedContext(Context)
+     */
+    public static Context getOrCreateDuplicatedContext(Vertx vertx) {
+        Vertx actual = Assert.checkNotNullParam("vertx", vertx);
+        Context context = actual.getOrCreateContext(); // Creates an event loop context if none
+        if (isDuplicatedContext(context)) { // Also checks that the context is not null
+            return context;
+        } else {
+            return ((ContextInternal) context).duplicate();
+        }
+    }
+
+    /**
+     * Checks if the given context is a duplicated context.
+     *
+     * @param context the context, must not be {@code null}
+     * @return {@code true} if the given context is a duplicated context, {@code false} otherwise.
+     */
+    public static boolean isDuplicatedContext(Context context) {
+        Context actual = Assert.checkNotNullParam("context", context);
+        return !(actual instanceof EventLoopContext || actual instanceof WorkerContext);
+    }
+
+    /**
+     * Checks if the current context is a duplicated context.
+     * If the method is called from a Vert.x thread, it retrieves the current context and checks if it's a duplicated
+     * context. Otherwise, it returns false.
+     *
+     * @return {@code true} if the method is called from a duplicated context, {@code false} otherwise.
+     */
+    public static boolean isOnDuplicatedContext() {
+        Context context = Vertx.currentContext();
+        return context != null && isDuplicatedContext(context);
+    }
+
```

Comment:
I think we would need a couple more methods :
- `@Nullable Context createDuplicatedContext()` : if on vertx context, create new duplicate context, otherwise null
- `@Nullable Context getOrCreateDuplicatedContext()` : if on vertx context, getOrCreateDuplicate, otherwise null
- `Context parentContext(Context context)` : return unwrap if duplicated context, otherwise itself

Reply:
Methods added (with slightly different names)